src/odemis/dataio/tiff.py

```diff
@@ -1727,6 +1733,23 @@ def _reconstructFromOMETIFF(xml, data, basename):
 
     return omedata
 
+def _gen_resized_shapes(original_shape):
+    """
+    Generates a list of tuples with the size of the resized images
+    original_shape (tuple of shape YX or YXC): shape of the original image
+    return (list of tuples): List of the tuples with the size of the resized images
+    """
+    shape = original_shape
+    resized_shapes = []
+    z = 0
+    while shape[0] >= TILE_SIZE and shape[1] >= TILE_SIZE:
```

Comment:
Actually, it's most important here to support array with any dimension. It should return list of shapes the same size as the input array.
=> Take the DataArray as argument (instead of just the shape).
=> To get the dimensions use `dims = data.metadata.get(model.MD_DIMS, "CTZYX"[-data.ndim:])` 
=> `while shape[dims.index("X")] >= TILE_SIZE and shape[dims.index("Y")] >= TILE_SIZE`
=> `shape = tuple(s // 2**z if d in "XY" else s for s,d in zip(orshape, dims))`